### ISSUE TITLE
[bug] password not removed from clipboard

### ISSUE THREAD

**Issue by stefanb2** [bkueng/qMasterPassword#34]
The copied password is not removed from the clipboard once the timer expires. As far as I remember that has never worked for me.

**NOTE:** I'm using the program in a KDE Plasma 5 Wayland session, so Wayland may be the root cause.

**Comment by bkueng** [comment]
I'm using it on X11, which works, so I guess it's indeed Wayland related (I know, it's time to switch to Wayland)

**Comment by stefanb2** [comment]
After looking at the code I have to revise my initial assumption. This has nothing to do with Wayland.

* call `QApplication::clipboard()` to access the `QClipboard` instance
* fetch the current clipboard text with `old = QClipboard::text()` and remember it
* copy the password or username to the clipboard with `QClipboard::setText(new)`
* after the timeout expires, check if the current clipboard is still the text copied into it and then reset the old content with `QClipboard::setText(old)`

As far as the clipboard is concerned, this approach is correct, as there is only one entry in the clipboard.

But most users have a clipboard manager on their desktop, e.g. the default Klipper on KDE Plasma. These manage a clipboard history, i.e. a call to `QClipboard::setText()` *pushes a new entry on top* of the history stack (and in Klipper's case it will remove duplicate entries from the stack).

I did a quick search and there doesn't seem to be a XDG API for managing clipboard history. So the only way to "fix" this would be to implement something that talks to the clipboard manager API directly, if it has one. For Klipper that would be the DBus interface, but that doesn't offer a way to clear specific history entries.

Closing as **WONTFIX**
